### PR TITLE
Feat/scrum 49 make script for autoencoder

### PR DESCRIPTION
**Pull request by Shawn-Kim96**
Made `dataset_preprocess/preprocess_function` to simplify data preprocessing by using module.

@Rodarkhen You can add more feature_engineering functions to this preprocess_function

### CODE REVIEW

File: src/dataset_preprocess/preprocess_functions.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        self.climate_columns = data.select_dtypes(include='number').columns.tolist()
+        self.autoencoder = None
+        self.scaler = None
+        self.preprocessed_data = None
+        self.process_and_fill_date_column()
+
+    def process_and_fill_date_column(self, date_col='date'):
+
+        Process a mixed-format date column, extract date features,
+        and fill missing values in those features.
```
Comment (hashemJaber) [inline src/dataset_preprocess/preprocess_functions.py:24]: I am not sure if this is an error but did you mean to comment out all the code below ?
or was it meant to be a commend line for "Process a mixed-format"

File: src/dataset_preprocess/preprocess_functions.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        self.encoder = nn.Sequential(
+            nn.Linear(input_dim, 64),
+            nn.ReLU(),
+            nn.Linear(64, 32),
+            nn.ReLU()
+        )
+        self.decoder = nn.Sequential(
+            nn.Linear(32, 64),
+            nn.ReLU(),
+            nn.Linear(64, input_dim)
```
Comment (hashemJaber) [inline src/dataset_preprocess/preprocess_functions.py:76]: This is a correct AutoEncoder 😄 , output is input_dim and not 1 as I saw in https://github.com/Shawn-Kim96/California_Wild_Fire_Prediction/blob/f876d66ed699c762652fad5e8f90c8ffe5ee668e/notebooks/ken/model.ipynb?short_path=f06808b#L247C5-L253C19

one issue is that for the course this won't be accepted since this is no longer a feature preprocessing step as you may be inadvertently be changing the dimensional representation of the feature vector, unless losses chosen shows a highly low loss.

**Review by hashemJaber — CHANGES_REQUESTED** [review]
